.github/workflows/build.yaml

```diff
@@ -83,7 +83,7 @@ jobs:
       - name: Workaround checkout Needed single revision issue
         run: git submodule foreach 'git rev-parse HEAD > /dev/null 2>&1 || rm -rf $PWD' || true
 
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```

Comment:
I'm just a random bystander, and have no ability to merge, but out of idle curiosity, what are the implications and consequences of changing from v3 to v4?

Reply:
https://github.com/actions/checkout

Self-hosted runners must be able to run Node 20 (GLIBC ≥ 2.28 on Linux). GitHub migrates all official actions to Node 20, so you will have to upgrade in the medium term

Node 16 was deprecated at the end of 2023, but v3 was still using it.

GitHub will show warnings if you are still using v3 when Node 16 disappears.